### DISCUSSION TITLE
How to Implement Database Transactions in Appwrite?

### DISCUSSION THREAD

**Discussion by Yavnik** [appwrite/appwrite#5078] [Q&A]
Hello Everyone,

I am working on a project where Atomicity of data is compulsory.
That is, whenever I want to update a bunch of documents (could be tens or even hundreds), I want to ensure either all of them updates or none at all and thus the database isn't changed.

Typically this would be implemented using Transactions where you commit if everything goes alright else rollback to previous state.

However, Appwrite currently does not support Transactions ( Issue #2788 ) and my project cannot continue without a solid implementation of Transactions. It's kind of a deal breaker.

So I'm looking at alternative methods to ensure atomicity using only the methods provided by Appwrite i.e. getDocument(), listDocuments(), updateDocument(), deleteDocument().

1) Any ideas on a way to ensure Atomicity and implement COMMIT or ROLLBACK using above methods only? (Maybe 2 Phase Commit?)
2) Any update on the progress of Issue #2788 ?

Any help is much appreciated.
Thanks.

**Comment by stnguyen90** [comment]
One workaround could be to use an Appwrite Function and execute it asynchronously since async executions are serial. You may also need to ensure the Function is the only thing updating the data.
